test/integration/features/proxy.feature

```diff
@@ -0,0 +1,81 @@
+@proxy @minishift-only
+Feature: Minishift can run behind proxy
+  As a user I can use Minishift behind a proxy.
+  INFO: Code behind this feature will start a proxy server in background
+        on port 8181 and will try to get IP of host machine automatically
+        to set MINISHIFT_HTTP_PROXY for this feature only.
+        However one can override port number or set IP manually if needed.
+        To use custom port number run this test with environmental variable
+        INTEGRATION_PROXY_CUSTOM_PORT.       
+        To use custom IP address please set INTEGRATION_PROXY_CUSTOM_IP
+        environmental variable.
+
+  Scenario: Start behind the proxy
+   Given user starts proxy server and sets MINISHIFT_HTTP_PROXY variable
+    When executing "minishift start" succeeds
+    Then proxy log should contain "Accepting CONNECT to registry-1.docker.io:443"
+     And proxy log should contain "Accepting CONNECT to auth.docker.io:443"
+
+  Scenario: User can login to the server
+   Given Minishift has state "Running"
+    When executing "oc login --username=developer --password=developer" succeeds
+    Then stdout should contain
+     """
+     Login successful
+     """
+
+  Scenario: User can create new namespace ruby for application ruby-ex
+   Given Minishift has state "Running"
+    When executing "oc new-project ruby" succeeds
+    Then stdout should contain
+     """
+     Now using project "ruby"
+     """
+
```

Comment:
@hferentschik So it seems that for non-interactive shell the environmental variables are not loaded from /var/lib/boot2docker/profile:
```
$ minishift ssh -- echo $HTTP_PROXY

$
```

Thinking if this is an issue or just normal behaviour? Can be here an user case which would require non-interactive shell and will need all parts of profile executed? I know that this is quite common around .bash_rc or .bash_profile, not sure here - can be resolved somehow, but for some files it is just known and expected to behave like this.  

For now I think it could be just fine to do:
```cucumber
  Scenario: Proxy environmental variable is set in created VM
    When executing "minishift ssh -- cat /var/lib/boot2docker/profile" succeeds
    Then stdout should match "HTTP_PROXY=http\:\/\/\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}\:\d{1,5}"
     And stdout should match "NO_PROXY=localhost,127.0.0.1,172.30.1.1"
```